emukit/core/acquisition/integrated_acquisition.py

```diff
@@ -29,12 +29,15 @@ def __init__(self, model: Union[IModel, IPriorHyperparameters], acquisition_gene
         self.step_size = step_size
         self.leapfrog_steps = leapfrog_steps
 
-        self.samples = self.model.generate_hyperparameters_samples(n_samples, n_burnin,
-                                                                   subsample_interval, step_size, leapfrog_steps)
+        self.update_parameters()
 
         acquisition = self.acquisition_generator(model)
         self._has_gradients = acquisition.has_gradients
 
+    @property
+    def samples(self):
+        return self.model.samples
+
```

Comment:
This acquisition seems to be the only place that actually makes use of samples as a part of its state.

Reply:
Yes. I've moved "samples" to be stored here instead, and moved the content of GPyModelWrapper.generate_hyperparameters_samples into IntegratedHyperParameterAcquisition.update_parameters, as the latter was only calling the former and was handing it a lot of attributes from IntegratedHyperParameterAcquisition - so it really belongs in that class.